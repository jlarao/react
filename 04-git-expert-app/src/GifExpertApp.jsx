import { useState } from "react";
import { AddCategory, GifGrid } from "./components";
export const GifExpertApp = () => {
    const [categories, setCategories] =  useState(['Dragon Ball']);
    
    const onAddCategory = ( newCategory) => {
        if (categories.includes(newCategory)) {
            alert('Ya existe esta categoria');
            return;
        }
        setCategories([newCategory, ...categories]);
    }
    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory 
                // onsetCategories={setCategories}
                onNewCategory={onAddCategory}
                />
            
            {
                categories.map(category => (
                    <GifGrid 
                    key={category} 
                    category={category}
                    />
                ))
            }
                
            
        </>
    );
}