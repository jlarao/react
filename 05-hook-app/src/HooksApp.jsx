

export function HooksApp() {

    return (
        <div>
            <h1>Hook App</h1>
        </div>
    )
}