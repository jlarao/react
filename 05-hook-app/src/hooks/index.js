export * from './UseCounter';
export * from './UseForm';
export * from './useFetch';