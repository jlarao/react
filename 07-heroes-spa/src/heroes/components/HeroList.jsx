import { HeroCard } from "./";
import { getHeroesByPublisher } from "../helpers"
import { useCallback, useEffect, useMemo, useState } from "react";

export const HeroList = ({ publisher }) => {
    
    // const heroes = getHeroesByPublisher(publisher);
    const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);
    

    // console.log(heroes);
    return (
        <div className="row row-cols-1 row-cols-md-3 g-3">
            {
                heroes.map(hero => (
                    <HeroCard key={hero.id} {...hero} />
                ))
            }
        </div>
    )
}