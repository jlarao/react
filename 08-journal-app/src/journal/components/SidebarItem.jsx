import { TurnedInNot } from "@mui/icons-material"
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useMemo } from "react"
import { useDispatch, useSelector } from "react-redux"
import { setActiveNote } from "../../store/Journal"

export const SidebarItem = ({title ='', body, id, date, imageUrls =[] }) => {
    // const { setActiveNote} = useSelector( state => state.journal);
    const dispatch = useDispatch();
    const onClicNote = () =>{
        // console.log('click note', id);
        dispatch(setActiveNote({id, title, body, date, imageUrls}));
    }
    const newTitle = useMemo( () => {
        // console.log(Title);
        return title.length > 17 ? title.substring(0, 17) + '...'
            : title   }, [title] );
    return(
    <ListItem disablePadding>
        <ListItemButton onClick={ onClicNote}>
            <ListItemIcon>
                <TurnedInNot/>
            </ListItemIcon>
        
        
        <Grid container>
            <ListItemText primary={ newTitle }/>
            <ListItemText secondary={ body}/>
        </Grid>
        </ListItemButton>
    </ListItem>
    )
}