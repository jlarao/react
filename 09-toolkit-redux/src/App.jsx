import { useState } from 'react'
import './App.css'
import { useDispatch, useSelector } from 'react-redux'
import { increment, decrement, incrementByAmount } from './store/slices/counter';

function App() {

  const {counter} = useSelector(state => state.counter);
  const dispatch  = useDispatch();

  return (
    <>
      
      <h1>Vite + React</h1>
      <div className="card"> 
        <h2>{counter}</h2>
        <button onClick={ () => { dispatch(increment())} }>
          increment
        </button>

        <button onClick={ () => { dispatch(decrement())}}>
            Decrement
        </button>

        <button onClick={ () => { dispatch(incrementByAmount(2))}}>
            Increment by 2
        </button>
        
      </div>
      
    </>
  )
}

export default App
