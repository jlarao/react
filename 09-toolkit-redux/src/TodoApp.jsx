import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis"

export const TodoApp = () => {

    // const { isLoading, data: todos = []} = useGetTodosQuery();
    const [todoid, setTodoid] = useState(1);
    const {  data: todo, isLoading } = useGetTodoQuery(todoid);

    const nextTodo = () => {
        setTodoid(todoid + 1);
    }

    const prevTodo = () => {
        if(todoid === 1) return;
        setTodoid(todoid -1);
    }
// console.log(todo);
    return (
        <>
            <h1>TodoApp RTK Query</h1>
            <hr />
            <h4>isLoading {isLoading ? 'True' : 'False'} </h4>

            <pre>{ JSON.stringify(todo) }</pre>
            {/* <ul>
                { todos.map( todo => 
                    (
                        <li key={ todo.id }> <strong>{todo.completed ? '✅' : '❌'}</strong> { todo.title } </li>
                    )
                ) }
            </ul> */}
            <button onClick={prevTodo}>Prev Todo</button>
            <button onClick={nextTodo}>Next Todo</button>
        </>
    )
}