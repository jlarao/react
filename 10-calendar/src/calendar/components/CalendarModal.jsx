import { addHours } from "date-fns";
import { act, useEffect, useMemo, useState } from "react";
import DatePicker, { registerLocale } from "react-datepicker";
import Modal from "react-modal";

import "react-datepicker/dist/react-datepicker.css";
import es from "date-fns/locale/es";

import Swal from "sweetalert2";
import  'sweetalert2/dist/sweetalert2.all.min.js'
import { swal } from "sweetalert2/dist/sweetalert2.all.min.js";
import { useCalendarStore, useUiStore } from "../../hooks";

registerLocale("es", es);

const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
    },
  };

export const CalendarModal = () => {
    const  { activeEvent, startSavingEvent } = useCalendarStore()

    //   const [isOpen, setIsOpen] = useState(true);
      const [formSubmitted, setFormSubmitted] = useState(false);

        const[formValues, setFormValues] =   useState({
            title: '',
            notes: '',
            start: new Date(),
            end: addHours( new Date(),2),
        });

        const titleClass = useMemo(() => {
            if(!formSubmitted) return '';

            return (formValues.title.length > 0) ? 'is-valid' : 'is-invalid';

        }, [formValues.title, formSubmitted]);

        useEffect(() => {
            if(activeEvent !== null) {
                setFormValues({...activeEvent})
            }
        }, [activeEvent]);
        const onInputChange = (e) => {
            setFormValues({
                ...formValues,
                [e.target.name]: e.target.value
            })
        }

        const onDateChange = (e, changing) => {
            setFormValues({
                ...formValues,
                [changing]: e
            })
        }
                

      Modal.setAppElement('#root');

      const { isDateModalOpen, closeDateModal } = useUiStore();
      const onCloseModal = () => {
        //   setIsOpen(false);
        closeDateModal();
        console.log('closed');
      }

      onsubmit = async(e) => {

          setFormSubmitted(true);
          e.preventDefault();
          console.log(formValues);
          const difference = formValues.end - formValues.start;

          if(NaN === (difference) || difference <= 0) {
            Swal.fire('Error', 'la fecha fin debe ser mayor a la fecha de inicio', 'error');
              console.log('la fecha fin debe ser mayor a la fecha de inicio');
              return;
          }

          if(formValues.title.length <= 0) {
            Swal.fire('Error', 'el evento debe tener un titulo', 'error');
              console.log('el evento debe tener un titulo');
              return;
          }

          await startSavingEvent( formValues );
        //   setIsOpen(false);
          console.log('saved');
          closeDateModal();

      }

    return (
        <Modal
        isOpen={ isDateModalOpen }
        onRequestClose={onCloseModal}
        style={customStyles}
        className="modal"
        overlayClassName="modal-fondo"
        closeTimeoutMS={200}
        
        >
            <h1> Nuevo evento </h1>
            <hr />
            <form className="container" onSubmit={onsubmit}>

                <div className="form-group mb-2">
                    <label>Fecha y hora inicio</label>
                    <DatePicker 
                    selected={formValues.start} 
                    onChange={ (e) => onDateChange(e, 'start') } 
                    name="start" 
                    className="form-control"
                    dateFormat="Pp"
                    showTimeSelect  
                    locale={'es'}
                    />
                </div>

                <div className="form-group mb-2">
                    <label>Fecha y hora fin</label>
                    <DatePicker 
                    minDate={formValues.start}
                    selected={formValues.end} 
                    onChange={ (e) => onDateChange(e, 'end') } 
                    name="end" 
                    className="form-control"
                    dateFormat="Pp"
                    showTimeSelect
                    locale={'es'} 
                    />
                </div>

                <hr />
                <div className="form-group mb-2">
                    <label>Titulo y notas</label>
                    <input 
                        type="text" 
                        className={`form-control ${titleClass}`}
                        placeholder="Título del evento"
                        name="title"
                        autoComplete="off"
                        value={formValues.title}
                        onChange={ onInputChange }
                    />
                    <small id="emailHelp" className="form-text text-muted">Una descripción corta</small>
                </div>

                <div className="form-group mb-2">
                    <textarea 
                        type="text" 
                        className="form-control"
                        placeholder="Notas"
                        rows="5"
                        name="notes"
                        autoComplete="off"
                        value={formValues.notes}
                        onChange={ onInputChange }
                    ></textarea>
                    <small id="emailHelp" className="form-text text-muted">Información adicional</small>
                </div>

                <button
                    type="submit"
                    className="btn btn-outline-primary btn-block"
                >
                    <i className="far fa-save"></i>
                    <span> Guardar</span>
                </button>

            </form>
        </Modal>
    )
}