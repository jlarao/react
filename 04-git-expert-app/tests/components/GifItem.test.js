import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Test en <GifItem.js>', () => {
    const title = "Dragon Ball";
    const url = "https://dragon-ball-super.fandom.com/wiki/Dragon_Ball_Super";
    
    test('match con el Snapshot', () => {

        const { container} = render(
        <GifItem 
            title={title} 
            url ={url} 
            />);

        render(<GifItem title={title} url ={url} />);

        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title= {title} url={url} />);
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title= {title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })


})