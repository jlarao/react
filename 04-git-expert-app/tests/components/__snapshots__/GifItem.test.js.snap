// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem.js> match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://dragon-ball-super.fandom.com/wiki/Dragon_Ball_Super"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
